build/release/teamcity-make-and-publish-build.sh

```diff
@@ -11,14 +11,25 @@ build/builder.sh make .buildinfo/tag
 build_name="${TAG_NAME:-$(cat .buildinfo/tag)}"
 release_branch="$(echo "$build_name" | grep -Eo "^v[0-9]+\.[0-9]+")"
 
-bucket="${BUCKET-cockroach-builds}"
+if [[ -z "${DRY_RUN}" ]] ; then
+  bucket="${BUCKET-cockroach-builds}"
+  google_credentials=$GOOGLE_COCKROACH_CLOUD_IMAGES_CREDENTIALS
+  gcr_repository="us.gcr.io/cockroach-cloud-images/cockroach"
+  git_tag="${build_name}"
+else
+  bucket="${BUCKET:-cockroach-release-test}"
+  google_credentials="$GOOGLE_COCKROACH_RELEASE_CREDENTIALS"
+  gcr_repository="us.gcr.io/cockroach-release/cockroach-test"
+  git_tag="dryrun-${build_name}"
```

Comment:
I created alternate locations for pushing S3 artifacts and docker images. For pushing git tags, I chose to push to `cockroachlabs/release-staging` rather than create a new repo (partly because cockroachlabs is at its limit of 125 private repos). 

What do you think of this approach?

To use DRY_RUN with the "Mark Build As X" build configurations, that script will need to remove the `dryrun-` prefix from the tag before using it.

Reply:
using same git remote makes sense to me. I'm not even sure we should `dry-run` the git tag?